src/rhsm/connection.py

```diff
@@ -635,24 +635,35 @@ def _print_debug_info_about_request(self, request_type, handler, final_headers,
             else:
                 msg = blue_col + "Making request:" + end_col
             msg += red_col + " %s:%s %s %s" % (self.host, self.ssl_port, request_type, handler) + end_col
+            if self.proxy_hostname and self.proxy_port:
+                msg += blue_col + " using proxy " + red_col + f"{self.proxy_hostname}:{self.proxy_port}" + end_col
             if 'SUBMAN_DEBUG_PRINT_REQUEST_HEADER' in os.environ:
                 msg += blue_col + " %s" % final_headers + end_col
             if 'SUBMAN_DEBUG_PRINT_REQUEST_BODY' in os.environ and body is not None:
                 msg += yellow_col + " %s" % body + end_col
             print()
             print(msg)
             print()
-            if os.path.isdir('/tmp/sub-man') is False:
-                os.mkdir('/tmp/sub-man')
             if 'SUBMAN_DEBUG_SAVE_TRACEBACKS' in os.environ:
-                with tempfile.NamedTemporaryFile(
-                        dir='/tmp/sub-man',
-                        mode='w',
-                        prefix='traceback-',
-                        delete=False) as tmp_file:
-                    traceback.print_stack(file=tmp_file.file)
-                    print(green_col + '    traceback saved in: %s' % tmp_file.name + end_col)
-                    print()
+                debug_dir = Path("/tmp/rhsm/")
+                debug_dir.mkdir(exist_ok=True)
+
+                timestamp: str = datetime.datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+
+                # make sure we don't overwrite previous logs
+                i = 0
+                while True:
+                    filename = Path(f"{timestamp}_{i}.log")
+                    debug_log = debug_dir / filename
+                    if not debug_log.exists():
+                        break
+                    i += 1
```

Comment:
I'd not bother with avoiding collisions. The pattern used now includes the seconds, which means it is extremely unlikely that two sub-man executions doing network communication will end within the same second.

Reply:
I didn't see any clashes myself. But I decided to play it safe: with local candlepin server the traffic will be much faster compared to the stage/prod one (and with stage I'm getting series of requests right after each other almost every time), and people living in US will also observe shorter response times.

And the last thing you want when debugging network traffic is log files overwriting each other.

Reply:
Fast as it can be, still getting runs within 1 second is very uncommon IMHO.

Also, you have always the option to... copy/move the old log file somewhere else, if needed ;)

Reply:
I think it's possible, if it happens to be at the start and at the end of the second. How would you copy the old log file, if it is overwritten in under a second with new content?

I wrote a simple loop that is overwrite-proof with minimal impact. Other way to solve the risk of overwriting files would be to include milliseconds (just the first decimal), would that be better?

Reply:
While I agree the chances of collision are low I think playing it safe helps to ensure we are able to access clean logs no matter the timing.